recipe/generator.go

```diff
@@ -0,0 +1,54 @@
+package recipe
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"text/template"
+)
+
+type FromTemplateData struct {
+	FileName string                 `json:"FileName" yaml:"FileName"`
+	Data     map[string]interface{} `json:"Data" yaml:"Data"`
+}
+
+type FromTemplateConfig struct {
```

Comment:
not really essential but could the name be just
```suggestion
type TemplateConfig struct {
```
?

Same with the one above as `TemplateData`